lib/absinthe/phase/document/execution/resolution.ex

```diff
@@ -75,11 +78,35 @@ defmodule Absinthe.Phase.Document.Execution.Resolution do
   end
 
   defp run_callbacks(plugins, callback, acc, true) do
-    Enum.reduce(plugins, acc, &apply(&1, callback, [&2]))
+    Enum.reduce(plugins, acc, fn plugin, acc ->
+      start_time = System.monotonic_time()
+
+      emit_start_event(start_time, acc, plugin)
+      acc = apply(plugin, callback, [acc])
+      emit_stop_event(start_time, acc, plugin)
+
+      acc
+    end)
   end
 
   defp run_callbacks(_, _, acc, _), do: acc
 
+  defp emit_start_event(start_time, acc, loader) do
+    :telemetry.execute(
+      @callback_start_event,
+      %{start_time: start_time},
+      %{acc: acc, loader: loader}
+    )
+  end
+
+  defp emit_stop_event(start_time, acc, loader) do
+    :telemetry.execute(
+      @callback_stop_event,
+      %{duration: System.monotonic_time() - start_time},
+      %{acc: acc, loader: loader}
```

Comment:
The `loader` variable here is I think the `plugin` right? We probably want to call it `plugin: plugin`